lib/presentation/util/toast_message_handler.dart

```diff
@@ -0,0 +1,204 @@
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+//
+
+
+import 'dart:async';
+
+import 'package:domain/domain.dart';
+import 'package:linshare_flutter_app/presentation/di/get_it_service.dart';
+import 'package:linshare_flutter_app/presentation/localizations/app_localizations.dart';
+import 'package:linshare_flutter_app/presentation/redux/actions/my_space_action.dart';
+import 'package:linshare_flutter_app/presentation/redux/actions/network_connectivity_action.dart';
+import 'package:linshare_flutter_app/presentation/redux/actions/share_action.dart';
+import 'package:linshare_flutter_app/presentation/redux/actions/shared_space_action.dart';
+import 'package:linshare_flutter_app/presentation/redux/actions/upload_file_action.dart';
+import 'package:linshare_flutter_app/presentation/redux/states/app_state.dart';
+import 'package:linshare_flutter_app/presentation/redux/states/my_space_state.dart';
+import 'package:linshare_flutter_app/presentation/redux/states/network_connectivity_state.dart';
+import 'package:linshare_flutter_app/presentation/redux/states/share_state.dart';
+import 'package:linshare_flutter_app/presentation/redux/states/shared_space_state.dart';
+import 'package:linshare_flutter_app/presentation/redux/states/upload_file_state.dart';
+import 'package:linshare_flutter_app/presentation/util/app_toast.dart';
+import 'package:redux/redux.dart';
+
+import 'package:flutter/material.dart';
+
+class ToastMessageHandler {
+  final Store<AppState> _store = getIt<Store<AppState>>();
+  final appToast = getIt<AppToast>();
+  StreamSubscription _streamSubscription;
+
+  void setup(BuildContext context) {
+    _streamSubscription = _store.onChange.listen((event) {
+      _handleMySpaceToastMessage(context, event.mySpaceState);
+      _handleUploadToastMessage(context, event.uploadFileState);
+      _handleShareDocumentToastMessage(context, event.shareState);
+      _handleSharedSpaceToastMessage(context, event.sharedSpaceState);
+      _handleNetworkStateToastMessage(context, event.networkConnectivityState);
+    });
+  }
+
+  void _handleMySpaceToastMessage(BuildContext context, MySpaceState mySpaceState) {
+    mySpaceState.viewState.fold((failure) {
+      if (failure is CopyToSharedSpaceFailure || failure is CopyMultipleFilesToSharedSpaceAllFailureViewState) {
+        appToast.showErrorToast(AppLocalizations.of(context).cannot_copy_file_to_shared_space);
+        _cleanMySpaceViewState();
```

Comment:
please find the way to move out all function `_cleanMySpaceViewState`. May be using either again

Reply:
but 1 question: this class is Toast handler, it is good when we clean state here?

Reply:
how about clean all the state in the obsever of each viewModel?

Reply:
agree, clean it in ViewModel seems better

Reply:
oh but we have some cases that toast is showed after the ViewModel is destroyed, for example: upload toast message is showed after upload screen is closed, so we can't observe there anymore. So I think we should handle it in `ToastMessageHandler`, WDYT ?

Reply:
Since My Space State contains only files list we can reset it when switching views, however account state must not be reset

Reply:
yes, we should reset unnecessary state when jump to other screens

Reply:
ok with @manhduc-nguyen solution, we will improve it in the future